apps/reference/docs/guides/platform/logs.mdx

```diff
@@ -4,78 +4,177 @@ title: Logging
+### LIMIT and Result Row Limitations
 
-#### Timestamp Behavior
+The Logs Explorer has a maximum of 1000 rows per run. To optimize your queries, you may wish to add in a `LIMIT` to your SQL query, to reduce the number of rows returned even further.
 
-Each log entry is stored with a `timestamp`. In order to utilize the `timestamp` field in a query, you can use the appropriate [timestamp functions](https://cloud.google.com/bigquery/docs/reference/standard-sql/timestamp_functions#timestamp).
+### Best Practices
 
-:::note
+1. Have a WHERE statement filtering over **timestamp**
 
-In the Log Explorer, timestamps are rendered as unix microsecond timestamps. SQL queries, however, should always use the `TIMESTAMP` data type. If you are using a unix timestamp value in a query, cast the value to a `TIMESTAMP` data type.
+  Querying across your entire log retention range might seem appealing, especially when you just need to get your query results fast. However, for **Enterprise** customers that have a large retention range, you would run the risk of your queries timing out due to the length cruch time required to scan and analyze across your large dataset. We recommend optimizing queries based on timestamp filtering, to ensure
 
-:::
 
-## Templates
+2. Avoid selecting large nested objects. Select individual values instead.
+
+  When querying large object, the columnar storage engine (in this case BigQuery) will select each column associated with each nested key, resulting in a large number of columns being selected. This inadvertently impacts the query speed and may result in timeouts or memory errors, especially for projects with large amount of logging activity. 
+
+  Instead, try to select only the values required, and where this is not possible, drill down to the minimally required object keys.
+
+  ```sql
+  -- ❌ edge_logs metadata contains many nested keys
+  SELECT timestamp, m as metadata
+  FROM edge_logs t
+  CROSS JOIN UNNEST(t.metadata) as m;
+  
+  -- ✅ select only the required values
+  SELECT timestamp, r.method
+  FROM edge_logs t
+  CROSS JOIN UNNEST(t.metadata) as m
+  CROSS JOIN UNNEST(m.request) as r
+  ```
+
+### Examples and Templates
+
+For example, you may enter the following into the SQL editor to query for each user's IP address:
+
+```sql
+SELECT DATETIME(timestamp), h.x_real_ip
+FROM edge_logs
+  LEFT JOIN UNNEST(metadata) as m ON TRUE
+  LEFT JOIN UNNEST(m.request) AS r ON TRUE
+  LEFT JOIN UNNEST(r.headers) AS h ON TRUE
+WHERE h.x_real_ip IS NOT NULL AND r.method = "GET"
+```
+
+Within the Logs Explorer, we have also included some templates to help you get started on each. Templates are available under the **Templates** tab, or under the **Templates** Dropdown in the **Query** tab.
```

Comment:
````suggestion
The Logs Explorer includes **Templates** (available in the Templates tab or the dropdown in the Query tab) to help you get started.

For example, you can enter the following query in the SQL Editor to retrieve each user's IP address:

```sql
SELECT DATETIME(timestamp), h.x_real_ip
FROM edge_logs
  LEFT JOIN UNNEST(metadata) as m ON TRUE
  LEFT JOIN UNNEST(m.request) AS r ON TRUE
  LEFT JOIN UNNEST(r.headers) AS h ON TRUE
WHERE h.x_real_ip IS NOT NULL AND r.method = "GET"
```
````